tests/cephadm/test_client.py

```diff
@@ -175,6 +175,9 @@ def setup(host):
 
             # Install ceph-common
             if config.get("install_packages"):
+                # Added this condition for upstream build
+                if config.get("ceph_docker_registry") == "quay.ceph.io":
```

Comment:
This is not ideal as it creates a dependency on a non-related parameter. This will run irrespective of `tentacle` version.

Reply:
^ Unless of course we make changes to pipeline to propagate the build info as an argument to run.py execution.

Actually, in retrospect, we should ideally be passing `--build upstream` in every upstream run regardless of whether `--upstream-build` is identified or repo and image details are provided separately. Without setting `--build upstream`, the method `def setup_upstream_repository` will never be invoked.

Maybe outside the scope of this PR, but this should also be addressed.

Reply:
> So the only way to do a conditional check for Upstream Tentacle would be to ensure rhbuild starts with 9 and --docker-registry is quay.ceph.io

I disagree with this statement.

We can introduce `--custom-config test_client_override_patch_openssl=true` and evaluate the same inside `test_client.py`. At the same time introduce this option to metadata in the pipelines of the specified upstream projects.

This way it is a single change point when the fix is introduced. Looking at `9` and `quay.ceph.io` in the long run is a bit difficult and does not provide any scope of extension

The primary purpose of `--custom-config` was to allow test configurations to be overriden outside of the test data.

Reply:
> We can introduce --custom-config test_client_override_patch_openssl=true and evaluate the same inside test_client.py. At the same time introduce this option to metadata in the pipelines of the specified upstream projects. 
> This way it is a single change point when the fix is introduced. Looking at 9 and quay.ceph.io in the long run is a bit difficult and does not provide any scope of extension

Understood, I assumed that we are not open to making changes in the metadata/pipeline for this workaround, that's why the inefficient suggestion.
Using `--custom-config test_client_override_patch_openssl=true` sounds reasonable, thanks

Reply:
thanks @harshkumarRH tested your solution worked fine
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-2HVH6X/